CMakeLists.txt

```diff
@@ -80,14 +80,21 @@ set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
 
 if(NOT USE_PREBUILT_LLVM)
     set(TARGET_BRANCH "ocl-open-90")
-    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+
+    if(NOT LLVM_EXTERNAL_CLANG_SOURCE_DIR)
+      set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+    elseif(EXISTS "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}/CMakeLists.txt") #monorepo
+      set(CLANG_SOURCE_DIR "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}")
+    endif()
```

Comment:
What if  `LLVM_EXTERNAL_CLANG_SOURCE_DIR` is set, but points to a wrong subdirectory? I think we should add `else` branch which emits error message

Reply:
Fixed